plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/web/aspects/ScrollToElement.java

```diff
@@ -0,0 +1,38 @@
+package ru.sbtqa.tag.pagefactory.web.aspects;
+
+import org.aeonbits.owner.ConfigFactory;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.openqa.selenium.*;
+import ru.sbtqa.tag.pagefactory.context.PageContext;
+import ru.sbtqa.tag.pagefactory.web.properties.WebConfiguration;
+
+@Aspect
+public class ScrollToElement {
+
+    private static final WebConfiguration properties = ConfigFactory.create(WebConfiguration.class);
+
+    @Pointcut("call(* org.openqa.selenium.WebElement.click()) && if()")
+    public static boolean isScrollToElementEnabled() {
+        return properties.isScrollToElementEnabled();
+    }
+
+    @Around("isScrollToElementEnabled()")
+    public void highlight(ProceedingJoinPoint joinPoint) throws Throwable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)